esphome/components/ld2450/ld2450.h

```diff
@@ -0,0 +1,287 @@
+  void set_zone_target_count_sensor(int zone, sensor::Sensor *s);
+  void set_zone_still_target_count_sensor(int zone, sensor::Sensor *s);
+  void set_zone_moving_target_count_sensor(int zone, sensor::Sensor *s);
+#endif
+  void reset_radar_zone();
+  void set_radar_zone(int32_t zone_type, int32_t zone1_x1, int32_t zone1_y1, int32_t zone1_x2, int32_t zone1_y2,
+                      int32_t zone2_x1, int32_t zone2_y1, int32_t zone2_x2, int32_t zone2_y2, int32_t zone3_x1,
+                      int32_t zone3_y1, int32_t zone3_x2, int32_t zone3_y2);
+
+ protected:
+  ESPPreferenceObject pref_;
+  int two_byte_to_int_(char firstbyte, char secondbyte) { return (int16_t) (secondbyte << 8) + firstbyte; }
+  void send_command_(uint8_t command_str, const uint8_t *command_value, int command_value_len);
+  void set_config_mode_(bool enable);
+  void handle_periodic_data_(uint8_t *buffer, int len);
+  bool handle_ack_data_(uint8_t *buffer, int len);
+  void process_zone_(uint8_t *buffer);
+  void readline_(int readch, uint8_t *buffer, int len);
+  void get_version_();
+  void get_mac_();
+  void query_zone_();
+  void restart_();
+  void send_set_zone_command_();
+  void convert_int_values_to_hex_(const int *values, uint8_t *bytes);
+  void save_to_flash_(float value);
+  float restore_from_flash_();
+  Target target_info_[MAX_TARGETS];
+  Zone zone_config_[MAX_ZONES];
+  int16_t decode_coordinate_(uint8_t low_byte, uint8_t high_byte) {
+    int16_t coordinate = (high_byte & 0x7F) << 8 | low_byte;
+    if ((high_byte & 0x80) == 0)
+      coordinate = -coordinate;
+    return coordinate;  // mm
+  }
+  int16_t decode_speed_(uint8_t low_byte, uint8_t high_byte) {
+    int16_t speed = (high_byte & 0x7F) << 8 | low_byte;
+    if ((high_byte & 0x80) == 0)
+      speed = -speed;
+    return speed * 10;  // mm/s
+  }
+  std::string convert_signed_int_to_hex_(int value) {
+    std::stringstream ss;
+    ss << std::hex << std::setw(4) << std::setfill('0') << (value & 0xFFFF);
+    return ss.str();
+  }
+  int16_t hex_to_signed_int_(const uint8_t *buffer, uint8_t offset) {
+    uint16_t hex_val = (buffer[offset + 1] << 8) | buffer[offset];
+    int16_t dec_val = static_cast<int16_t>(hex_val);
+    if (dec_val & 0x8000)
+      dec_val -= 65536;
+    return dec_val;
+  }
+  float calculate_angle_(float base, float hypotenuse) {
+    if (base < 0.0 || hypotenuse <= 0.0)
+      return 0.0;
+    float angle_radians = std::acos(base / hypotenuse);
+    float angle_degrees = angle_radians * (180.0 / M_PI);
+    return angle_degrees;
+  }
+  std::string get_direction_(int16_t speed) {
+    static const std::string APPROACHING = "Approaching";
+    static const std::string MOVING_AWAY = "Moving away";
+    static const std::string STATIONARY = "Stationary";
+
+    if (speed > 0)
+      return MOVING_AWAY;
+    if (speed < 0)
+      return APPROACHING;
+    return STATIONARY;
+  }
+  int32_t uptime_millis_ = millis();
+  int32_t last_periodic_millis_ = millis();
```

Comment:
Should not use millis here
 
```suggestion
  int32_t uptime_millis_ = 0;
  int32_t last_periodic_millis_ = 0;
```